src/status_im/contexts/wallet/wallet_connect/events.cljs

```diff
@@ -0,0 +1,71 @@
+(ns status-im.contexts.wallet.wallet-connect.events
+  (:require [re-frame.core :as rf]
+            [status-im.constants :as constants]
+            status-im.contexts.wallet.wallet-connect.effects
+            [status-im.contexts.wallet.wallet-connect.utils :as wallet-connect.utils]
+            [taoensso.timbre :as log]))
+
+(rf/reg-event-fx
+ :wallet-connect/init
+ (fn []
+   {:fx [[:effects.wallet-connect/init
+          {:on-success #(rf/dispatch [:wallet-connect/on-init-success %])
+           :on-fail    #(rf/dispatch [:wallet-connect/on-init-fail %])}]]}))
+
+(rf/reg-event-fx
+ :wallet-connect/on-init-success
+ (fn [{:keys [db]} [web3-wallet]]
+   {:db (assoc db :wallet-connect/web3-wallet web3-wallet)
+    :fx [[:dispatch [:wallet-connect/register-event-listeners]]]}))
+
+(rf/reg-event-fx
+ :wallet-connect/register-event-listeners
+ (fn [{:keys [db]}]
+   (let [web3-wallet (get db :wallet-connect/web3-wallet)]
+     {:fx [[:effects.wallet-connect/register-event-listener
+            web3-wallet
+            constants/wallet-connect-session-proposal-event
+            #(rf/dispatch [:wallet-connect/on-session-proposal %])]]})))
+
+(rf/reg-event-fx
+ :wallet-connect/on-init-fail
```

Comment:
We can clear the proposal from db if initialisation fails. Stale state is bad for health.

Reply:
agreed, but resetting the proposal should happen when "sessionApproved" fails/succeeds, so added it there.